src/points/types.ts

```diff
@@ -56,6 +61,12 @@ type SwapClaimHistory = BaseClaimHistory & {
 
 export type ClaimHistory = CreateWalletClaimHistory | SwapClaimHistory
 
+// See https://stackoverflow.com/questions/59794474/omitting-a-shared-property-from-a-union-type-of-objects-results-in-error-when-us
+type DistributiveOmit<T, K extends keyof any> = T extends any ? Omit<T, K> : never
```

Comment:
nit: Should we move this to [src/utils/typescript.ts](https://github.com/valora-inc/wallet/blob/main/src/utils/typescript.ts)?